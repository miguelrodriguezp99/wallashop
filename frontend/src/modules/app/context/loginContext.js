import { createContext, useState } from "react";

// Este es el que tenemos que consumir
export const LogginContext = createContext();

// Este es el que nos provee de acceso al contexto
export function LogginProvider({ children }) {
	const [token, setToken] = useState(null);
	const [image, setImage] = useState(null);
	const [user, setUser] = useState({});

	return (
		<LogginContext.Provider
			value={{
				token,
				setToken,
				image,
				setImage,
				user,
				setUser,
			}}
		>
			{children}
		</LogginContext.Provider>
	);
}
