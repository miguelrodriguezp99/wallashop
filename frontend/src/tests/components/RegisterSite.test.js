import React from 'react';
import RegisterSite from '../../modules/app/RegisterSite';
import renderer from 'react-test-renderer';
import { LogginContext } from "../../modules/app/context/loginContext"; // AsegÃºrate de usar la ruta correcta
import { MemoryRouter } from "react-router-dom";

import '@testing-library/jest-dom';

jest.mock('../../modules/app/firebase/config.js', () => ({
}));


const fakeLogginContext = {
	token: "testToken",
};

describe('RegisterSite component', () => {
    it('renders correctly', () => {
      const tree = renderer.create(
        <MemoryRouter>
            <LogginContext.Provider value={fakeLogginContext}>
                <RegisterSite />
            </LogginContext.Provider>
        </MemoryRouter>)
        .toJSON();
      expect(tree).toMatchSnapshot();
    });
});