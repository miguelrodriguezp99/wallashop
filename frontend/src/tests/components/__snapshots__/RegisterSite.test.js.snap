// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RegisterSite component renders correctly 1`] = `
<div
  className="flex flex-col items-center justify-center min-h-screen bg-black"
>
  <div
    className="border border-solid border-gray-200 rounded-2xl text-center text-gray-200 font-helvetica py-10 px-16 w-500 mt-0 hover:border-blue-300"
  >
    <h1
      className="text-7xl font-bold bg-gradient-to-r from-blue-500 to-blue-700 text-transparent bg-clip-text"
    >
      SIGN UP
    </h1>
    <form
      onSubmit={[Function]}
    >
      <label
        htmlFor="username"
      >
        <div
          className="form__group field"
        >
          <input
            className="form__field"
            name="username"
            onChange={[Function]}
            placeholder="Username"
            required=""
            type="text"
            value=""
          />
          <label
            className="form__label"
            htmlFor="name"
          >
            Username
          </label>
        </div>
      </label>
      <label
        htmlFor="email"
      >
        <div
          className="form__group field"
        >
          <input
            className="form__field"
            name="email"
            onChange={[Function]}
            placeholder="Email"
            required=""
            type="text"
            value=""
          />
          <label
            className="form__label"
            htmlFor="name"
          >
            Email
          </label>
        </div>
      </label>
      <label
        htmlFor="password"
      >
        <div
          className="form__group field"
        >
          <input
            className="form__field"
            name="password"
            onChange={[Function]}
            placeholder="Password"
            required=""
            type="password"
            value=""
          />
          <label
            className="form__label"
            htmlFor="name"
          >
            Password
          </label>
        </div>
      </label>
      <label
        htmlFor="firstname"
      >
        <div
          className="form__group field"
        >
          <input
            className="form__field"
            name="firstName"
            onChange={[Function]}
            placeholder="FirstName"
            required=""
            type="text"
            value=""
          />
          <label
            className="form__label"
            htmlFor="name"
          >
            FirstName
          </label>
        </div>
      </label>
      <label
        htmlFor="lastname"
      >
        <div
          className="form__group field"
        >
          <input
            className="form__field"
            name="lastname"
            onChange={[Function]}
            placeholder="LastName"
            required=""
            type="text"
            value=""
          />
          <label
            className="form__label"
            htmlFor="name"
          >
            LastName
          </label>
        </div>
      </label>
      <label
        className="mt-0 inline-block dark:text-neutral-200 text-white"
        htmlFor="formFile"
      >
        Avatar
      </label>
      <input
        className="relative m-0 block w-full min-w-0 flex-auto rounded border border-solid border-neutral-300 bg-clip-padding px-3 py-[0.32rem] text-base font-normal text-neutral-700 transition duration-300 ease-in-out file:-mx-3 file:-my-[0.32rem] file:overflow-hidden file:rounded-none file:border-0 file:border-solid file:border-inherit file:bg-neutral-100 file:px-3 file:py-[0.32rem] file:text-neutral-700 file:transition file:duration-150 file:ease-in-out file:[border-inline-end-width:1px] file:[margin-inline-end:0.75rem] hover:file:bg-neutral-200 focus:border-primary focus:text-neutral-700 focus:shadow-te-primary focus:outline-none dark:border-neutral-600 dark:text-neutral-200 dark:file:bg-neutral-700 dark:file:text-neutral-100 dark:focus:border-primary mt-5"
        id="formFile"
        onChange={[Function]}
        type="file"
      />
      <div
        className="form-buttons mt-5"
      >
        <button
          className="group tap-highlight-transparent transition-transform-colors-opacity data-[hover=true]:opacity-hover z-0 group relative inline-flex items-center justify-center box-border appearance-none select-none whitespace-nowrap font-normal subpixel-antialiased overflow-hidden tap-highlight-transparent outline-none data-[focus-visible=true]:z-10 data-[focus-visible=true]:outline-2 data-[focus-visible=true]:outline-focus data-[focus-visible=true]:outline-offset-2 border-medium bg-transparent px-unit-4 min-w-unit-20 h-unit-10 text-small gap-unit-2 rounded-medium [&>svg]:max-w-[theme(spacing.unit-8)] data-[pressed=true]:scale-[0.97] transition-transform-colors-opacity motion-reduce:transition-none border-primary hover:!text-primary-foreground hover:!bg-primary text-white ml-2 mr-2"
          disabled={false}
          href="#"
          onBlur={[Function]}
          onClick={[Function]}
          onDragStart={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchCancel={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          role="button"
          tabIndex={0}
          type="button"
        >
          Sign Up
        </button>
        <button
          className="group tap-highlight-transparent transition-transform-colors-opacity data-[hover=true]:opacity-hover z-0 group relative inline-flex items-center justify-center box-border appearance-none select-none whitespace-nowrap font-normal subpixel-antialiased overflow-hidden tap-highlight-transparent outline-none data-[focus-visible=true]:z-10 data-[focus-visible=true]:outline-2 data-[focus-visible=true]:outline-focus data-[focus-visible=true]:outline-offset-2 border-medium bg-transparent px-unit-4 min-w-unit-20 h-unit-10 text-small gap-unit-2 rounded-medium [&>svg]:max-w-[theme(spacing.unit-8)] data-[pressed=true]:scale-[0.97] transition-transform-colors-opacity motion-reduce:transition-none border-primary hover:!text-primary-foreground hover:!bg-primary text-white ml-2 mr-2"
          disabled={false}
          href="#"
          onBlur={[Function]}
          onClick={[Function]}
          onDragStart={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchCancel={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          role="button"
          tabIndex={0}
          type="button"
        >
          Home
        </button>
      </div>
    </form>
  </div>
</div>
`;
